src/libpriv/rpmostree-util.c

```diff
@@ -605,6 +605,34 @@ rpmostree_cache_branch_to_nevra (const char *cachebranch)
   return g_string_free (r, FALSE);
 }
 
+/* Implementation taken from https://git.gnome.org/browse/libgsystem/tree/src/gsystem-log.c */
+gboolean
+rpmostree_stdout_is_journal (void)
+{
+  static gsize initialized;
+  static gboolean stdout_is_socket;
+
+  if (g_once_init_enter (&initialized))
+    {
+      guint64 pid = (guint64) getpid ();
+      g_autofree char *fdpath = g_strdup_printf ("/proc/%" G_GUINT64_FORMAT "/fd/1", pid);
```

Comment:
Hmm, reading the systemd docs, it looks like it gives us a safe way to check this: https://www.freedesktop.org/software/systemd/man/systemd.exec.html#%24JOURNAL_STREAM. So this becomes something like:

```c
stdout_is_journal = FALSE;
const char *journal_stream = getenv ("JOURNAL_STREAM");
if (journal_stream != NULL)
  {
    struct stat stbuf;
    if (fstat (STDOUT_FILENO, &stbuf) != -1)
      {
        g_autofree char *expected = g_strdup_printf ("%u:%u", stbuf.st_dev, stbus.st_ino);
        stdout_is_journal = g_str_equal (expected, journal_stream);
      }
  }

g_once_init_leave (&initialized, TRUE);
```
? (Just exercising my C here :))

I'm surprised it doesn't provide a library function that does the above for us.

Reply:
CentOS/RHEL systemd is too old to have that (just try `systemd-run env`), and even for Fedora, as the docs say we'd have to check anyways so...we'd basically do the same thing whether or not it's set right?

Reply:
Though we could start collecting these things in a `git grep`-able way like:

`TODO: CentOS7 EOL: use JOURNAL_STREAM`?

Then [in 2024](https://access.redhat.com/support/policy/updates/errata) maybe we can remember to grep for it? 😄

Reply:
Ahh, drats CentOS/RHEL! Meh, 2024 seems too far to warrant a `TODO` for something like this. :)

> and even for Fedora, as the docs say we'd have to check anyways so...we'd basically do the same thing whether or not it's set right?

I think the docs just say that the existence of `JOURNAL_STREAM` is necessary but not sufficient, i.e. we actually need to `fstat` stdout/err and compare. If `JOURNAL_STREAM` is not defined, then we should just assume we're not outputting to the journal?